ui/main/src/app/services/processes.service.ts

```diff
@@ -222,4 +222,65 @@ export class ProcessesService {
         };
     }
 
+    public findServiceForProcess(processId : string) : string {
+        for (let group of this.processGroups) {
+            if (group.processes.find(process => process == processId))
+                return group.id;
+        }
+        return '';
+    }
+
+    public makeProcessesDropdownListPerServices(): Map<any, any> {
+        let processesDropdownListPerServices = new Map();
+
+        this.getAllProcesses().forEach(process => {
+
+            const service = this.findServiceForProcess(process.id);
+            if (service != '') {
+                let processes = (!! processesDropdownListPerServices.get(service) ? processesDropdownListPerServices.get(service) : []);
+                processes.push({id: process.id, itemName: process.name, i18nPrefix: `${process.id}.${process.version}`});
+                processesDropdownListPerServices.set(service, processes);
+            }
+        });
+        return processesDropdownListPerServices;
+    }
+
+    public makeProcessesWithoutServiceDropdownList(): any[] {
+        let processesWithoutServiceDropdownList = [];
+
+        this.getAllProcesses().forEach(process => {
+            const service = this.findServiceForProcess(process.id);
+            if (service == '')
+                processesWithoutServiceDropdownList.push({ id: process.id, itemName: process.name, i18nPrefix: `${process.id}.${process.version}` });
+        });
+        return processesWithoutServiceDropdownList;
+    }
+
+    public loadAllProcessesPerServices(object: {processesDropdownListPerServices: Map<any, any>;
```

Comment:
Why not doing it once on init of the service as the result will always be the same 

So I would prefer having three structure pre-calculates and 3 methods to get access : 
- getListOfServices()
- getListOfProcessPreService()
- getListOfProcessWithoutService() 

having the construction of dropDownList outside of this service as it is more related to UI consideration

Reply:
> Why not doing it once on init of the service as the result will always be the same
> 
> So I would prefer having three structure pre-calculates and 3 methods to get access :
> 
>     * getListOfProcessGroup()
A function getProcessGroups() already exists in the file.

> 
>     * getListOfProcessPerProcessGroup()
Done : function getProcessesPerProcessGroups()
> 
>     * getListOfProcessWithoutProcessGroup()
Done : function getProcessesWithoutProcessGroup()
> 
> 
> having the construction of dropDownList outside of this service as it is more related to UI consideration

Reply:
Done. commit 79ed090dad53132056f4e5652caea5f514ed3f01